### PR TITLE
[DEV-5161] Removing dead TAS, NAICS, and PSC Code

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
Removes the code from the old filters.

**Technical details:**
- TAS: The old TAS filter had two tabs, one of which we're deleting, the other which we are keeping. I had to change some of the logic and some of the logic in the tests.

Besides the main containers/components for each filter, the following pages needed to have code deleted for each filter:

- **Models**: 
	- SearchAwardsOperation
- **Reducers**: 
	- searchFiltersReducer
	- otherFilterFunctions
- **Components**
	- topFilterGroupGenerator
- **Containers**
	- TopFilterBarContainer
- **DataMapping**
	- awardsOperationKeys
- **Tests**
	- mockSearchHashes
- **SCSS**

NOTE: Did not touch any SCSS.

COMMITS:
PSC: 2c0cbe3 & 61fe3dc
NAICS: c9a064e
TAS: de3d89b

**JIRA Ticket:**
[DEV-5161](https://federal-spending-transparency.atlassian.net/browse/DEV-5161)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
`N/A` Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
`N/A` Verified cross-browser compatibility
`N/A` Verified mobile/tablet/desktop/monitor responsiveness
`N/A` Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
`N/A` Added Unit Tests for methods in Container Components, reducers, and helper functions `if applicable`
`N/A` All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)
`N/A` [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
`N/A` Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/js/components/search/topFilterBar/filterGroups/NAICSFilterGroup.jsx [hunk 1]
```diff
@@ -37,23 +37,13 @@ export default class NAICSFilterGroup extends React.Component {
...
-                    title: `${naics.naics} | ${naics.naics_description}`,
-                    isSpecial: false,
-                    removeFilter: this.removeFilter
-                };
-            });
+            .map((naics) => ({
+                value: `${naics.identifier}`,
+                title: `${naics.value} - ${naics.label} (${naics.count})`,
+                // doesn't appear to be being used...
+                isSpecial: false,
```
Comment (ebdabbs) [inline src/js/components/search/topFilterBar/filterGroups/NAICSFilterGroup.jsx:44]: Agree, we can go ahead and remove the `isSpecial` prop

File: src/js/components/search/topFilterBar/filterGroups/PSCFilterGroup.jsx [hunk 2]
```diff
@@ -37,22 +37,12 @@ export default class PSCFilterGroup extends React.Component {
...
-                value: `${value.identifier}`,
-                title: `${value.product_or_service_code} - ${value.psc_description}`,
-                isSpecial: false,
-                removeFilter: this.removeFilter
-            };
-        });
+        return this.props.filter.values.map((value) => ({
+            value: `${value.value}`,
+            title: `${value.psc_description}`,
+            isSpecial: false,
```
Comment (ebdabbs) [inline src/js/components/search/topFilterBar/filterGroups/PSCFilterGroup.jsx:43]: Can be removed here too

File: tests/containers/search/filters/naics/mockNAICS.js [hunk 3]
```diff
@@ -1,706 +1,806 @@
...
+                naics: "children_of_11",
+                isPlaceHolder: true,
+                count: 0
             }
         ]
     }
 ];
 
-export const naicsMockCleanData = [
+export const treeWithPlaceholdersAndRealDataPSCDepth = [
```
Comment (ebdabbs) [inline tests/containers/search/filters/naics/mockNAICS.js:118]: Is this for testing NAICS, PSC, or both? Looks like a depth of 4 but NAICS fields. Maybe add a comment to explain.

File: tests/redux/reducers/search/searchFiltersReducer-test.js [hunk 4]
```diff
@@ -710,39 +644,7 @@ describe('searchFiltersReducer', () => {
...
-                selectedPSC: new OrderedMap({
-                    [pscNum]: expectedPSC
-                })
-            });
-
-            const updatedState = searchFiltersReducer(startingState, action);
-            expect(updatedState.selectedPSC).toEqual(new OrderedMap());
-        });
-    });
+    // TODO: Add Test for New PSC Reducer
```
Comment (ebdabbs) [inline tests/redux/reducers/search/searchFiltersReducer-test.js:647]: Should we do this in this PR?

File: src/js/components/search/topFilterBar/filterGroups/NAICSFilterGroup.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/components/search/topFilterBar/filterGroups/NAICSFilterGroup.jsx]: 199ee87b2


☝️ touches 24 files 😱 .

All of em though just call the same component. Seems like a really inefficient way of rendering this markup!

File: src/js/components/search/topFilterBar/filterGroups/PSCFilterGroup.jsx (hunk 2 above)
Comment (MaxwellKendall) [reply src/js/components/search/topFilterBar/filterGroups/PSCFilterGroup.jsx]: 199ee87b2

File: tests/containers/search/filters/naics/mockNAICS.js (hunk 3 above)
Comment (MaxwellKendall) [reply tests/containers/search/filters/naics/mockNAICS.js:118]: Yeah I added a todo to remove this and replace it with PSC mock data.

File: tests/redux/reducers/search/searchFiltersReducer-test.js (hunk 4 above)
Comment (MaxwellKendall) [reply tests/redux/reducers/search/searchFiltersReducer-test.js:647]: Probably!!

**Pull request merged**